packages/kit/src/runtime/client/client.js

```diff
@@ -1220,6 +1220,13 @@ export function create_client({ target, session, base, trailing_slash }) {
 				}
 			});
 
+			// fix link hrefs, because browsers will occasionally try to load relative
+			// URLs after a pushState/replaceState, resulting in a 404 — see
+			// https://github.com/sveltejs/kit/issues/3748
+			for (const link of document.querySelectorAll('link')) {
+				link.href = link.href; // eslint-disable-line
```

Comment:
should we only do this if `link.href` doesn't contain `://`? it might be more performant to avoid updating the DOM and on later navigations most links will already be absolute

Reply:
i'm pretty sure `link.href = link.href` is just a no-op if it's already absolute. this only runs once (in `_start_router`), not on every navigation